src/main/java/soot/Scene.java

```diff
@@ -947,12 +959,13 @@ public Type getType(String arg) {
         result = VoidType.v();
       } else if (type.equals("boolean")) {
         result = BooleanType.v();
-      } else {
-        throw new RuntimeException("unknown type: '" + type + "'");
+      } else if (allowsPhantomRefs() && phantomNonExist) {
```

Comment:
Could you please document the meaning of phantomNonExist in the JavaDoc of this method?